### PR TITLE
eth/catalyst: adjust eta for themerge

### PR DESCRIPTION
**Pull request by holiman**
This PR changes the way we estimate time for the merge. There's a subtle error in the current algorithm

Consider the case where `2` headers are used for the estimation, rather than `64`. In that case, the difficulties for both blocks will be added, and the time between the two blocks. So the estimate will be based on `2 block difficulties / 1 block period`. 

So the current algo is based on `64 difficulties / 63 block periods`, over-estimating the difficulty per second.

Aside from that, I made it use `types.Header`s instead of blocks (easier on the db), got rid of some indentation, and made it never print out an estimate of `0` (which would happen in `PrettyAge` when time is below `1 second`.)

### CODE REVIEW

**Comment by holiman** [comment]
```
...before:
WARN [08-25|08:07:23.003] Merge is configured, but no beacon client seen. Please ensure you have one available before the transition arrives! eta=3w1d9h
WARN [08-25|08:12:23.033] Merge is configured, but no beacon client seen. Please ensure you have one available before the transition arrives! eta=3w2d5h
... after : 
WARN [08-25|08:13:17.108] Merge is configured, but no beacon client seen. Please ensure you have one available before the transition arrives! eta=3w1d19h
WARN [08-25|08:18:17.139] Merge is configured, but no beacon client seen. Please ensure you have one available before the transition arrives! eta=2w6d3h
```

**Review by karalabe — APPROVED** [review]
SGTM

File: eth/catalyst/api.go [hunk 1]
```diff
@@ -589,89 +589,84 @@ func (api *ConsensusAPI) heartbeat() {
...
+				if lastTransitionUpdate.IsZero() {
+					log.Warn("Merge already reached, but no beacon client seen. Please launch one to follow the chain!")
+				} else {
+					log.Warn("Merge already reached, but previously seen beacon client is offline. Please ensure it is operational to follow the chain!")
+				}
+				offlineLogged = time.Now()
+				continue
+			}
+			var eta time.Duration
+			if head.Number.Uint64() > 0 && htd.Cmp(ttd) < 0 {
```
Comment (karalabe) [inline eth/catalyst/api.go:639]: You don't need the `htd.Cmp(ttd) < 0` check any more, , the >= case was covered above

File: eth/catalyst/api.go [hunk 2]
```diff
@@ -589,89 +589,84 @@ func (api *ConsensusAPI) heartbeat() {
...
+				// Accumulate the last 64 difficulties to estimate the growth
+				var deltaDifficulty uint64
+				var deltaTime uint64
```
Comment (karalabe) [inline eth/catalyst/api.go:642]: Please combine the two var declarations into

```
var (
  ...
  ...
)
```

File: eth/catalyst/api.go [hunk 3]
```diff
@@ -589,89 +589,84 @@ func (api *ConsensusAPI) heartbeat() {
...
+				hdr := head
```
Comment (karalabe) [inline eth/catalyst/api.go:643]: Aslo add this, and perhaps rename "current"

File: eth/catalyst/api.go [hunk 4]
```diff
@@ -589,89 +589,84 @@ func (api *ConsensusAPI) heartbeat() {
...
+				} else {
+					log.Warn("Merge already reached, but previously seen beacon client is offline. Please ensure it is operational to follow the chain!")
+				}
+				offlineLogged = time.Now()
+				continue
+			}
+			var eta time.Duration
+			if head.Number.Uint64() > 0 && htd.Cmp(ttd) < 0 {
+				// Accumulate the last 64 difficulties to estimate the growth
+				var deltaDifficulty uint64
```
Comment (karalabe) [inline eth/catalyst/api.go:641]: Pls rename to deltaDiff, will look better aligned with deltaTime

**Pull request merged**